### DISCUSSION TITLE
integrate old pattern with new pattern

### DISCUSSION THREAD

**Discussion by knight6700** [pointfreeco/swift-composable-architecture#1607] [Q&A] [answered]
i want to integrate New Pattern TCA `enum ReducerProtocol` with old TCA  `let Reducer<State, Action, Environment>`
it can possible to do this

**Comment by stephencelis** [comment, marked as answer]
@knight6700 Have you looked at our article on migrating to the `ReducerProtocol`? It shows how the new protocol can be integrated in existing applications, and how to migrate existing reducer values to types conforming to the protocol: https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/migratingtothereducerprotocol
